map/src/main/kotlin/edu/artic/map/MapFragment.kt

```diff
@@ -0,0 +1,248 @@
+package edu.artic.map
+
+import android.graphics.Bitmap
+import android.os.Bundle
+import android.view.View
+import com.bumptech.glide.Glide
+import com.bumptech.glide.request.target.SimpleTarget
+import com.bumptech.glide.request.transition.Transition
+import com.fuzz.rx.disposedBy
+import com.google.android.gms.maps.GoogleMap
+import com.google.android.gms.maps.MapView
+import com.google.android.gms.maps.MapsInitializer
+import com.google.android.gms.maps.model.*
+import edu.artic.analytics.ScreenCategoryName
+import edu.artic.db.models.ArticGallery
+import edu.artic.db.models.ArticMapAnnotation
+import edu.artic.db.models.ArticObject
+import edu.artic.map.util.ArticObjectMarkerGenerator
+import edu.artic.map.util.DepartmentMarkerGenerator
+import edu.artic.map.util.GalleryNumberMarkerGenerator
+import edu.artic.viewmodel.BaseViewModelFragment
+import io.reactivex.android.schedulers.AndroidSchedulers
+import kotlinx.android.synthetic.main.fragment_map.*
+import kotlin.reflect.KClass
+
+class MapFragment : BaseViewModelFragment<MapViewModel>() {
+
+    override val viewModelClass: KClass<MapViewModel>
+        get() = MapViewModel::class
+    override val title: String
+        get() = ""
+    override val layoutResId: Int
+        get() = R.layout.fragment_map
+    override val screenCategory: ScreenCategoryName
+        get() = ScreenCategoryName.Map
+    lateinit var destroyableMapView: MapView
```

Comment:
yes ok I'll add. but basically, using kotlinx version of mapview it works everywhere except in ondestroy (then it's a null pointer exception)

Reply:
ah right. did you try before the `super.onDestroy()` call?

Reply:
yup.
still crashed

Reply:
probably because `onDestroyView()`. what about putting the `mapView.onDestroy()` there? im prety sure thats the proper place in a fragment anyways. since we dont want it around once the view is destroyed.